core/src/main/scala/codesearch/core/Main.scala

```diff
@@ -55,72 +57,61 @@ object Main {
     }
   }
 
-  case class LangRep[T <: DefaultTable](db: DefaultDB[T], langIndex: LanguageIndex[T])
-
   private val langReps = Map(
-    "haskell"    -> LangRep[HackageTable](HackageDB, HaskellIndex()),
-    "rust"       -> LangRep[CratesTable](CratesDB, RustIndex()),
-    "ruby"       -> LangRep[GemTable](GemDB, RubyIndex()),
-    "javascript" -> LangRep[NpmTable](NpmDB, JavaScriptIndex())
+    "haskell"    -> LangRep(HackageDB, HaskellIndex()),
+    "rust"       -> LangRep(CratesDB, RustIndex()),
+    "ruby"       -> LangRep(GemDB, RubyIndex()),
+    "javascript" -> LangRep(NpmDB, JavaScriptIndex())
   )
 
   def main(args: Array[String]): Unit = {
 
     parser.parse(args, Config()) foreach { c =>
-      if (c.lang != "all" && !(langReps.keySet contains c.lang)) {
+      if (c.lang != "all" && !(langReps.keySet contains c.lang))
         throw new IllegalArgumentException(s"Unsupported lanuages\n Available languages: ${langReps.keys}")
-      }
 
-      if (c.lang == "all") {
+      if (c.lang == "all")
         logger.info("Codesearch-core started for all supported languages")
-      } else {
+      else
         logger.info(s"Codesearch-core started for language ${c.lang}")
-      }
 
       if (c.initDB) {
         val future = c.lang match {
-          case "all" =>
-            Future.sequence(langReps.values.map(_.db.initDB()))
-          case lang =>
-            langReps(lang).db.initDB()
+          case "all" => Future.sequence(langReps.values.map(_.db.initDB()))
+          case lang  => langReps(lang).db.initDB()
         }
         Await.result(future, Duration.Inf)
       }
 
       if (c.downloadMeta) {
         c.lang match {
-          case "all" =>
-            langReps.values.foreach(_.langIndex.downloadMetaInformation())
-          case lang =>
-            langReps(lang).langIndex.downloadMetaInformation()
+          case "all" => langReps.values.foreach(_.langIndex.downloadMetaInformation())
+          case lang  => langReps(lang).langIndex.downloadMetaInformation()
         }
       }
 
       if (c.updatePackages) {
         val future = c.lang match {
           case "all" =>
             Future
-              .sequence(langReps.values.map(_.langIndex.updatePackages()))
```

Comment:
I think that scalafmt not support it. Because brackets depend from implementation and scalafmt can only format. If function without parameters is pure then can write without brackets, but if function have implicit parameter - brackets required.